packages/notte-sdk/src/notte_sdk/types.py

```
@@ -498,24 +498,32 @@ class NotteProxy(SdkRequest):
-            if "country" not in values and isinstance(geolocation, dict) and "country" in geolocation:
-                values["country"] = geolocation["country"]
+            if isinstance(geolocation, dict):
+                # Only set fields if they're not already provided.
+                if "country" not in values and "country" in geolocation:
+                    values["country"] = geolocation["country"]
+                if "city" not in values and "city" in geolocation:
+                    values["city"] = geolocation["city"]
         return values  # type: ignore[return-value]
 
     @staticmethod
     def from_country(country: str, id: str | None = None) -> "NotteProxy":
```

Ruff (0.15.14)

[error] Function argument `id` is shadowing a Python builtin

(A002)